### ISSUE TITLE
Add creating empty maven project without archetypes

### ISSUE THREAD

**Issue by sevapopov2** [microsoft/vscode-maven#869]
## Suggestion
Add an ability to create an empty maven project without archetypes like it can be done in IntelliJ IDEA.
## Use Cases
Quick creation of maven project without removing test sample files that were added by archetype afterwards and editing pom.xml file for convinsing the installed Java version.
## Examples
This behavior can be found in IntelliJ IDEA. You can create an empty maven project with folders structure and the correct Java version will be already specified in pom.xml file.

**Comment by Eskibear** [comment]
Technically, you can create your own archetype, without adding test samples. Then you can create whatever empty project as you like.

**Comment by Eskibear** [comment]
FYI, archetypes installed in your local catalog will always be listed on the first page, as implemented below.

https://github.com/microsoft/vscode-maven/blob/09cb9349da2a6beeca86bb2ce0b45497b0f235e7/src/archetype/createProject/SelectArchetypeStep.ts#L104-L105

**Comment by mbetrifork** [comment]
I'm an eclipse user converting to vscode and creating maven modules without any archetype is what I do 99% of the time. For me it would be a very useful feature to have.

Can you provide a reason for closing this feature with wontfix?

**Comment by Eskibear** [comment]
@mbetrifork 

> creating maven modules without any archetype is what I do 99% of the time

There is a simple solution. I can embed an "empty" sample maven project, and whenever you want to create, it's copied into target location. Here my question is, what should this empty project look like? A single pom.xml without any source files or you want me to create basic folders like `src/main/java`?

**Comment by Eskibear** [comment]
> Can you provide a reason for closing this feature with wontfix?

I closed it as wontfix because no one continued to provide any further feedback after waiting for several weeks. FYI

**Comment by mbetrifork** [comment]
I agree with @sevapopov2. In principle a new Maven module could be anything from a simple pom.xml file to a larger project structure suitable for a web application for example. But it is a good starting point to assume a new java project with src/main/java+resources and src/test/java+resources and a basic pom file.

It is also not very logical (to me :-) how Maven modules are created vs new Maven projects. This could be relevant in this context too, because the new pom file would have to reference the correct parent pom.

But basically I think @sevapopov2 has the right idea here.

**Comment by Eskibear** [comment]
OK, from the discussion above, I would spare some time working on this feature after I finish my current tasks. Basically I would embed a template of pom.xml, filling whatever needed like artifactId etc. according to user input. It would also create basic folder structures like src/main/java. You are also welcome to do this if you are interested in it, please let me know if so.

> It is also not very logical (to me :-) how Maven modules are created vs new Maven projects.

This somehow makes sense. Ideally, the to-be-implemented feature should be better called "Create new module", while a new project is still created from an archetype. Do you think that makes sense? Or we simply call it "create new empty project" or whatever.

**Comment by mbetrifork** [comment]
Eclipse and IntelliJ both allow the user to select an archetype or an "archetype-less" project regardless of the type of module being created (parent or child) and this principle is likely what most users expect to see applied in VS Code as well. It makes sense to me at least.

If it is known beforehand that the user is creating a parent project which will contain modules, you don't need an archetype selection because the parent is always a single pom file and no source directories.

Maybe it is possible to have two commands, "Maven: Create new parent project" (implicitly without an archetype) and "Maven: Create new project or module" which would allow for both archetype selection and "No archetype".

**Comment by sevapopov2** [comment]
After rereading the last comment, I think in case of creating a new project yes, there can be an option in archetype selection that will be called like "no archetype: create an empty maven project without an archetype". Something like that, so the user would know what it means. I think creation of resources folder in src/main/ would be also useful. We put different files to work there, and it might be useful for beginners who might not know that we put other files than java there by standard. For example, I learned about "Resources" folder only after using IntelliJ Idea, because it is only created there and not in VSCode. As far as projects with big amount of modules with parent projects, I cannot tell anything since I've never worked with such.

**Comment by Eskibear** [comment]
Thanks for your feedbacks. For simplicity, I want to start with adding an entry in current "Create Maven Project" command, creating a basic project as discussed above. 

Below is a demo of the workflow.

https://user-images.githubusercontent.com/2351748/231350474-3b01d68b-858d-446d-a556-51342cc192ae.mp4

**Comment by Eskibear** [comment]
The feature is merged and will be available in next nightly build (probably tomorrow). You can switch to pre-realease version to give it a try then.

See https://github.com/microsoft/vscode-maven/pull/962#issuecomment-1504695317 for how to use it.

**Comment by mbetrifork** [comment]
Just want to say thank you for your work on this. Looks good to me :-)

I can't tell from the video just what exactly is put into the new module pom file when there is no archetype but I can check that out myself.

**Comment by Eskibear** [comment]
> what exactly is put into the new module pom file 

It's almost the same with what you see in 0:44 of the video, with groupId/artifactId specified by your own. One main difference from using an artifact is, the newly created module won't be added into parent pom's `modules` section automatically, which might be implement in the future if it's critical.

**Comment by sevapopov2** [comment]
Huge thank you for implementing that!
Unfortunately I cannot see video and check since I am a blind user but I will be really happy to see how it works when the new extension will be available in usual release! Thank you again!

**Comment by ILyaCyclone** [comment]
Hello. I think I'm missing something? Is this generally available? I've fresh installed VSCode few days ago and I cannot see "No archetype" option in a list. 

VSCode 1.78.2
Maven for Java extension 0.41.0

**Comment by Eskibear** [comment]
@ILyaCyclone Looks like it's not released yet. You can switch to pre-release channel to try it for the moment. 
/cc @jdneo @testforstephen who will help to release a new version.

**Comment by sevapopov2** [comment]
Hello,
Thank you for releasing that!
Could you please add the following line to a POM file:
<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
It is included in IntelliJ IDEA's POM file when creating Maven project, I don't know how much it is important but if it is added than it seems like it is needed?
Thank you!
